.storybook/stories/Input/Input.tsx

```diff
@@ -18,24 +28,65 @@ export const Input: FC<InputProps> = ({
   placeholder = 'Placeholder',
   variant = 'default',
   disabled = false,
+  autoCapitalize = 'sentences',
   centerText = false,
   onChange,
+  onBlur,
+  min = 0,
+  max = 100,
 }) => {
   const [type, setType] = useState<InputProps['variant'] | 'active'>(variant);
-  const onFocus = useCallback(() => setType('active'), []);
-  const onBlur = useCallback(() => setType(variant), [variant]);
+
+  const onChangeInput = useCallback((value: string) => {
+    const newValue = keyboardType === 'default' ? value : getNumberValue(value);
+    onChange(newValue);
+  }, []);
+
+  const onFocusInput = useCallback(
+    (value: NativeSyntheticEvent<TextInputFocusEventData>) => {
+      setType('active');
+    },
+    [],
+  );
+
+  const onBlurInput = useCallback(
+    (value: NativeSyntheticEvent<TextInputEndEditingEventData>) => {
+      setType(variant);
+      onBlur?.(value.nativeEvent.text);
+    },
+    [variant],
+  );
+
+  const getNumberValue = (value: string): string => {
+    if (value === '') {
+      value = min.toString();
+    }
+    const parts = value.replace(',', '.').split('.').slice(0, 2);
+    parts[0] = parseInt(parts[0]).toString();
```

Comment:
What exactly is happening in this line? Why you are parsing the first part of the value and then casting it to a string again? It doesn't make much sense to me. If you add a `.join('.')` to a previous line, it should work correctly

Reply:
Leading `0` are deleting there.